src/components/Todo/common/Todo/Item/index.tsx

```diff
@@ -81,6 +87,91 @@ export const Item: FC<Props> = ({
     setItemText(items[itemIndex].name);
   }, [changeFocus, focus, itemIndex, items]);
 
+  useEffect(() => {
+    setItemText(items[itemIndex].name);
+  }, [items[itemIndex].name]);
+
+  const handleDelete = () => {
+    items.splice(itemIndex, 1);
```

Comment:
Before changing the items array. Create a copy of items to avoid reference changes directly. You can do something like this.
```
const itemsCopy = [...items];
itemsCopy.splice(itemIndex, 1);
setItemsCallback([...itemsCopy]);
```